src/relay/pass/type_solver.cc

```diff
@@ -412,10 +602,8 @@ void TypeSolver::MergeFromTo(TypeNode* src, TypeNode* dst) {
 }
 
 // Add equality constraint
-Type TypeSolver::Unify(const Type& dst, const Type& src, const NodeRef&) {
-  // NB(@jroesch): we should probably pass location into the unifier to do better
-  // error reporting as well.
-  Unifier unifier(this);
+Type TypeSolver::Unify(const Type& dst, const Type& src, const NodeRef& loc) {
```

Comment:
I am very glad to see the improved error reporting. I think it would be good to have a comment that loc is for error reporting

Reply:
It is commented in the user-visible interfaces.